pandas/core/internals/construction.py

```diff
@@ -615,6 +615,8 @@ def _extract_index(data) -> Index:
             elif is_list_like(val) and getattr(val, "ndim", 1) == 1:
                 have_raw_arrays = True
                 raw_lengths.append(len(val))
+            elif isinstance(val, np.ndarray) and getattr(val, "ndim", 1) > 1:
```

Comment:
after the isinstance check the getattr can check can just be `val.ndim > 1`

Reply:
amended